.github/workflows/pypi.yml

```
@@ -0,0 +1,70 @@
+# Publish archives to PyPI and TestPyPI using GitHub Actions
+name: Publish to PyPI
+on: [push, pull_request, workflow_dispatch]
+jobs:
+  build-artifact:
+    name: Build echoregions package
+    runs-on: ubuntu-20.04
+    if: github.repository == 'OSOceanAcoustics/echoregions'
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        # fetch all history so that setuptools-scm works
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@v4.7.0
+      with:
+        python-version: 3.9
+    - name: Install dependencies
+      run: python -m pip install setuptools wheel
+    - name: Build source and wheel distributions
+      run: |
+        python setup.py sdist bdist_wheel
+        echo ""
+        echo "Generated files:"
+        ls -lh dist/
+    - uses: actions/upload-artifact@v3
+      with:
+        name: releases
+        path: dist
+
+  test-built-dist:
+    name: Test echoregions package
+    needs: build-artifact
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+    - uses: actions/setup-python@v4.7.0
+      name: Install Python
+      with:
+        python-version: 3.9
+    - uses: actions/download-artifact@v3
+      with:
+        name: releases
+        path: dist
+    - name: List contents of built dist
+      run: |
+        ls -ltrh
+        ls -ltrh dist
+    - name: Publish to Test PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
```

```suggestion
      uses: pypa/gh-action-pypi-publish@release/v1
      with:
        repository-url: https://test.pypi.org/legacy/
```

Need to specify test pypi if you're doing against the test. Otherwise it defaults to the regular pypi.